modules/core/src/main/java/org/apache/synapse/config/xml/endpoints/AddressEndpointFactory.java

```diff
@@ -115,7 +119,16 @@ public EndpointDefinition createEndpointDefinition(OMElement elem) {
         }
 
         if (address != null) {
-            endpointDefinition.setAddress(address.getAttributeValue().trim());
+            String extractedAddress = address.getAttributeValue().trim();
+            if (extractedAddress.contains(SYSTEM_VARIABLE_PREFIX)) {
+                String extractedEnvVariableKey = extractedAddress.substring(extractedAddress.lastIndexOf(":") + 1);
+                String extractedEnvVariableValue = System.getenv(extractedEnvVariableKey);
+                log.info ("Environment variable " + extractedEnvVariableKey + " replaced with " +
+                        extractedEnvVariableValue);
+                endpointDefinition.setAddress(extractedEnvVariableValue);
+            } else {
+                endpointDefinition.setAddress(extractedAddress);
```

Comment:
Remove the no. of variables and remove the else condition

Reply:
Completed in commit: https://github.com/apache/synapse/pull/55/commits/8e51818d398bf9316873db0704ac359ca68e7b05